src/api/app/models/cloud/upload_job.rb

```diff
@@ -4,21 +4,55 @@ class UploadJob
     include ActiveModel::Model
     extend Forwardable
 
-    attr_accessor :user_upload_job, :backend_upload_job
-    validate :validate_jobs
+    attr_accessor :user_upload_job, :backend_upload_job, :params
+    validate :validate_jobs, :validate_filename, :validate_architecture, :validate_regions, :validate_virtualization_types, :validate_ami_name
     def_delegator :backend_upload_job, :id
 
     def self.create(user, params)
-      upload_job = new
+      upload_job = new(params: params)
+      return upload_job if upload_job.invalid?
       upload_job.backend_upload_job = Cloud::Backend::UploadJob.create(user, params)
       return upload_job unless upload_job.backend_upload_job.valid?
 
       upload_job.user_upload_job = user.upload_jobs.create(job_id: upload_job.id)
       upload_job
     end
 
+    def self.uploadable_file?(filename)
+      filename.end_with?('.raw.xz', '.vhdfixed.xz')
+    end
+
+    def self.uploadable_architecture?(architecture)
+      architecture == 'x86_64'
+    end
+
     private
 
+    def validate_filename
+      return if self.class.uploadable_file?(params[:filename])
+      errors.add(:base, "'#{params[:filename]}' is not a valid cloud format")
+    end
+
+    def validate_architecture
+      return if self.class.uploadable_architecture?(params[:arch])
+      errors.add(:base, "'#{params[:arch]}' is not a valid cloud format")
+    end
+
+    def validate_regions
+      return if ::Cloud::Ec2::Configuration::REGIONS.map(&:second).include?(params[:region])
```

Comment:
Can't we use the existing validation helpers here? Like
```ruby
validates :region, inclusion: { in: ::Cloud::Ec2::Configuration::REGIONS, message: 'is not a valid EC2 region' }
```

Reply:
See my other comment! Therefore we need first to introduce these attributes which I'm not sure if it makes sense as we only will use them for the validation. I mean, sure this would simplify the validations with the overhead of unused attributes.

Reply:
Okay, you convinced me. But they are used. At least in the validations;-)

Reply:
Sure in the validations. Sure your suggestion could be another way of handling it. However, as said this would make it necessary to introduce all these variables only for the validations and also to dismantle the hash. So I'm not sure what is the better approach here.

@evanrolfe what do you think?

Reply:
I dont have a strong opinion but maybe try having a quick go at implementing bjoern's suggestion and see if the code is better/more readable or maybe dismantling the hash makes it more complicated than its worth?